### PR TITLE
[SPARK-45605][CORE][SQL][SS][CONNECT][MLLIB][GRAPHX][DSTREAM][PROTOBUF][EXAMPLES] Replace `s.c.MapOps.mapValues` with `s.c.MapOps.view.mapValues`

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr replace `s.c.MapOps.mapValues` with `s.c.MapOps.view.mapValues`  due to `s.c.MapOps.mapValues` marked as deprecated since Scala 2.13.0:

https://github.com/scala/scala/blob/bf45e199e96383b96a6955520d7d2524c78e6e12/src/library/scala/collection/Map.scala#L256-L262

```scala
  @deprecated("Use .view.mapValues(f). A future version will include a strict version of this method (for now, .view.mapValues(f).toMap).", "2.13.0")
  def mapValues[W](f: V => W): MapView[K, W] = new MapView.MapValues(this, f)
```

### Why are the changes needed?
Cleanup deprecated API usage.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
- Pass GitHub Acitons
- Packaged the client, manually tested `DFSReadWriteTest/MiniReadWriteTest/PowerIterationClusteringExample`.

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

**Comment by LuciferYang** [comment]
Please wait while I manually validate the modifications to the examples.

**Comment by LuciferYang** [comment]
cc @HyukjinKwon @srowen FYI

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

File: common/sketch/src/test/scala/org/apache/spark/util/sketch/CountMinSketchSuite.scala [hunk 1]
```diff
@@ -56,7 +56,7 @@ class CountMinSketchSuite extends AnyFunSuite { // scalastyle:ignore funsuite
 
       val exactFreq = {
         val sampledItems = sampledItemIndices.map(allItems)
-        sampledItems.groupBy(identity).mapValues(_.length.toLong)
+        sampledItems.groupBy(identity).view.mapValues(_.length.toLong)
```
[the remark this answers is not in the document]
Comment (cloud-fan) [reply common/sketch/src/test/scala/org/apache/spark/util/sketch/CountMinSketchSuite.scala:59]: As an early Spark contributor, I can bet that when most people write `.mapValue`, they never consider this lazy thing but just treat it the same as `Seq.map`.

For example, this is a test, who cares if the map is lazy or not?

@LuciferYang can you help to go through all the changes in this PR, and identify the places that do need a lazy map?

**Pull request closed without merging**